src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt

```diff
@@ -31,18 +27,16 @@ class LicenceService(
     val curfew = licenceData.curfew
     val proposedAddress = licenceData.proposedAddress
 
-    return curfew?.curfewHours?.let { formatCurfewHoursObject(curfew.curfewHours) }?.let {
-      HdcLicence(
-        licenceId = licence.id,
-        curfewAddress = getAddress(curfew, cas2Referral, proposedAddress),
-        firstNightCurfewHours = curfew.firstNight,
-        // curfewHours referred to as curfewTimes in CVL as going forward a more suitable name and had to distinguish between the two different curfew data formats
-        curfewTimes = it,
-      )
-    }
+    return HdcLicence(
+      licenceId = licence.id,
+      curfewAddress = getAddress(curfew, cas2Referral, proposedAddress),
+      firstNightCurfewHours = curfew?.firstNight?.let { transformToModelFirstNight(it) },
+      // curfewHours referred to as curfewTimes in CVL as going forward a more suitable name and had to distinguish between the two different curfew data formats
+      curfewTimes = curfew?.let { transformToModelCurfewTimes(curfew.curfewHours) },
```

Comment:
not worth changing but can refer to curfew as `it` inside let.

Reply:
Good spot!